tools/ProductConstructionService.ReproTool/Operations/FlowCommitOperation.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.ProductConstructionService.Client;
+using Microsoft.DotNet.ProductConstructionService.Client.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using ProductConstructionService.ReproTool.Options;
+using GitHubClient = Octokit.GitHubClient;
+
+namespace ProductConstructionService.ReproTool.Operations;
+
+internal class FlowCommitOperation : Operation
+{
+    private readonly FlowCommitOptions _options;
+    private readonly ILogger<FlowCommitOperation> _logger;
+    private readonly GitHubClient _ghClient;
+    private readonly DarcProcessManager _darc;
+    private readonly IProductConstructionServiceApi _localPcsApi;
+
+    public FlowCommitOperation(
+            FlowCommitOptions options,
+            ILogger<FlowCommitOperation> logger,
+            GitHubClient ghClient,
+            DarcProcessManager darc,
+            [FromKeyedServices("local")] IProductConstructionServiceApi localPcsApi)
+        : base(logger, ghClient, localPcsApi)
+    {
+        _options = options;
+        _logger = logger;
+        _ghClient = ghClient;
+        _darc = darc;
+        _localPcsApi = localPcsApi;
+    }
+
+    internal override async Task RunAsync()
+    {
+        await _darc.InitializeAsync();
+
+        _logger.LogInformation("Flowing commit from {sourceRepo}@{sourceBranch} to {targetRepo}@{targetBranch}",
+            _options.SourceRepository,
+            _options.SourceBranch,
+            _options.TargetRepository,
+            _options.TargetBranch);
+
+        List<Channel> channels = await _localPcsApi.Channels.ListChannelsAsync();
+        Channel? channel = channels.FirstOrDefault(c => c.Name == _options.Channel)
+            ?? await _localPcsApi.Channels.CreateChannelAsync("test", _options.Channel);
+
+        var subscriptions = await _localPcsApi.Subscriptions.ListSubscriptionsAsync(
+            channelId: channel.Id,
+            sourceRepository: _options.SourceRepository,
+            targetRepository: _options.TargetRepository,
+            sourceEnabled: true);
+
+        var (repoName, owner) = GitRepoUrlParser.GetRepoNameAndOwner(_options.SourceRepository);
+
+        var isBackflow = false;
+        try
+        {
+            await _ghClient.Repository.Content.GetAllContents(owner, repoName, SourceMappingsPath);
+            isBackflow = true;
+        }
+        catch { }
+
+        var mappingName = (isBackflow ? _options.TargetRepository : _options.SourceRepository).TrimEnd('/').Split('/').Last();
+
+        Subscription subscription = subscriptions.FirstOrDefault(s => s.TargetBranch == _options.TargetBranch)
```

Comment:
why not pass the target branch above, where we're listing all subs

Reply:
There's no such parameter

Reply:
Ah ok